static/js/settings.js

```diff
@@ -330,6 +331,32 @@ const SettingsScreen = {
     });
   },
 
+  getAddonList: function() {
+    // If already fetched addon list, return a promise cached.
+    if (this.fetchAddonDeferred) {
+      return this.fetchAddonDeferred.catch(() => {
+        this.fetchAddonDeferred = null;
+        return this.getAddonList();
+      });
+    }
+
+    let resolveFunc = null, rejectFunc = null;
+
+    this.fetchAddonDeferred = new Promise((resolve, reject) => {
```

Comment:
This would be more clear:

```javascript
    this.fetchAddonDeferred = new Promise((resolve, reject) => {
      this.fetchAddonList().then(() => {
        resolve();
      }).catch((e) => {
        console.error(`Failed to parse add-ons list: ${e}`);
        reject(e);
      });
    });
```

Reply:
I get the point and addressed.